secboot/secboot.go

```diff
@@ -251,3 +253,50 @@ func MarkSuccessful() error {
 
 	return nil
 }
+
+// AuthMode corresponds to an authentication mechanism.
+type AuthMode string
+
+const (
+	AuthModePassphrase AuthMode = "passphrase"
+	// TODO: Add PIN option when secboot support lands.
+)
+
+// VolumesAuthOptions contains options for the volumes authentication
+// mechanism (e.g. passphrase authentication).
+type VolumesAuthOptions struct {
+	Mode       AuthMode `json:"mode,omitempty"`
+	Passphrase string   `json:"passphrase,omitempty"`
+	// TODO: Add PIN option when secboot support lands.
+	KDFType string        `json:"kdf-type,omitempty"`
+	KDFTime time.Duration `json:"kdf-time,omitempty"`
```

Comment:
btw. I think we do it in the notices already, but since we're expecting this struct in the daemon API, it exposes Go's semantics of duration to the client, which does not accept duration without units. I don't know if it's a problem, but the client now needs to be aware of Go's `time.ParseDuration` expectations.

Reply:
That's a very good point, My initial thought was this would be in nanoseconds? It seems that notices does `time.ParseDuration` manually. I tried json marshalling/unmarshalling and doesn't seem to accept strings. Could the json decorder be doing something different?

Reply:
But parsing as nanoseconds is expected, at least according to the SD201 spec nanoseconds are expected.

Reply:
I see, if it's in the spec and agreed on then it's fine